lib/customisation.nix

```diff
@@ -131,8 +131,9 @@ rec {
 
 
   /* Add attributes to each output of a derivation without changing
-     the derivation itself. */
-  addPassthru = drv: passthru:
+     the derivation itself and then map all outputs with given
+     functions. */
+  extendDerivation = passthru: f: d: o: drv:
```

Comment:
According to the name of the function, `drv` should be the first parameter.

None of the `f`, `d`, `o` use the first of the two arguments this function passes to them, so it should not pass it at all. They also simply return their second argument (what else can be done with an output path?), so it should not be passed either. The new function needs just one new argument `isValid` (or `{ isValid ? true; }`) to use as:
```
drvPath = assert isValid; drv.${outputName}.drvPath;
outPath = assert isValid; drv.${outputName}.outPath;
```